packages/eslint-plugin/src/rules/no-duplicate-enum-values.ts

```diff
@@ -0,0 +1,73 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'no-duplicate-enum-values',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow duplicate enum member values',
+      recommended: false,
+      suggestion: true,
+    },
+    hasSuggestions: true,
+    messages: {
+      duplicateMember: 'Duplicate enum member with value {{value}}.',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    function isStringLiteral(
+      node: TSESTree.Expression,
+    ): node is TSESTree.StringLiteral {
+      return (
+        node.type === AST_NODE_TYPES.Literal && typeof node.value === 'string'
+      );
+    }
+
+    function isNumberLiteral(
+      node: TSESTree.Expression,
+    ): node is TSESTree.NumberLiteral {
+      return (
+        node.type === AST_NODE_TYPES.Literal && typeof node.value === 'number'
+      );
+    }
```

Comment:
[Nitpicking] Is there a need to separate these two functions out? They both check that `node.type === AST_NODE_TYPES.Literal`.

I'd suggest either making a single separate function:

```ts
function getExpressionLiteralValue(node: TSESTree.Expression) {
  if (node.type === AST_NODE_TYPES.Literal) {
    switch (typeof node.value) {
      // ...
    }  
  }
}

...or inlining them below:

```ts
let value: string | number | undefined;

if (node.type === AST_NODE_TYPES.Literal) {
  // ...
}
```

Just suggestions, feel free to ignore if you don't like either of these changes 😄